packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts

```diff
@@ -326,6 +327,31 @@ export class GaussianSplattingMesh extends Mesh {
         super.dispose(doNotRecurse);
     }
 
+    /**
+     * Returns a new Mesh object generated from the current mesh properties.
+     * @param name is a string, the name given to the new mesh
+     * @returns a new Gaussian Splatting Mesh
+     */
+    public override clone(name: string = ""): GaussianSplattingMesh {
+        const newGS = new GaussianSplattingMesh(name, undefined, this.getScene());
+        newGS._copySource(this);
+        newGS._vertexCount = this._vertexCount;
+        newGS._modelViewMatrix = Matrix.Identity();
+        newGS._covariancesATexture = this._covariancesATexture;
+        newGS._covariancesBTexture = this._covariancesBTexture;
+        newGS._centersTexture = this._centersTexture;
+        newGS._colorsTexture = this._colorsTexture;
+        newGS._splatPositions = this._splatPositions;
+        newGS._instanciateWorker();

```

Comment:
A new worker is instanciated with a new thin instance buffer. But matrix of the original GS will also update indices of the clones. I don't see why, I need  another pair of eyes here!

Reply:
In materials, we clone the textures, but it is easy as they are not dynamic, and they end up reusing the same internal texture (ref count is done within the textures).

I guess we should do the same and try cloning the texture ?

Reply:
what does the worker update? does it make any changes to the positions?

Reply:
worker updates _depthMix that is then transfered to splatIndex that is used for thin instances. It doesn't change the positions but it sorts the indices based on transformed positions. So each worker needs a detached position buffer (am I correct here?)

Reply:
You can see the issue with this PG : https://playground.babylonjs.com/?snapshot=refs/pull/15292/merge#CID4NN#154 top is source GS, bottom is clone. source sorted array is overriden by the clone one (I least, it looks like that)

Reply:
It was not an issue with webworkers. It's fixed now.